### PR TITLE
Soften Profile inheritence rules

### PR DESCRIPTION
**Pull request by stain**
In some cases, such as [Workflow Run Crate](https://www.researchobject.org/workflow-run-crate/profiles/) there are multiple profiles that are related with inheritance.

There are also less strict inheritance profiles such as https://trefx.uk/trusted-wfrun-crate/0.3/

This section now makes it softer how to list multiple profiles, they are suggestions rather than enforcing any hierarchy.

### CODE REVIEW

**Review by simleo — CHANGES_REQUESTED** [review]

**Comment by ptsefton** [comment]
Didn't we decide to remove all this from 1.2? @stain

**Comment by ptsefton** [comment]
This needs to be moved to a different document, right?

**Comment by stain** [comment]
Remove inheritance, at best "referenced/suggested" profiles.

**Comment by stain** [comment]
Decision 2024-04-11: Not do formal inheritance system at RO-Crate level yet before we have tried out how it works with actual implementations (e.g. SHACL, Crate-O modes) -- then facilitate them.

File: docs/1.2-DRAFT/profiles.md [hunk 1]
```diff
@@ -470,4 +506,51 @@ it is [resolvable](appendix/jsonld.md#ro-crate-json-ld-media-type)
...
+    "@id": "https://w3id.org/ro/wfrun/process/0.1",
+    "@type": ["CreativeWork", "Profile"],
+    "name": "Process Run crate profile",
+    "version": "0.1.0"
+},
+{ 
+    "@id": "#shouldConformTo",
+    "@type": "ResourceDescriptor",
+    "hasRole": { "@id": "http://purl.org/dc/terms/conformsTo" },
+    "hasArtifact": {"@id": "https://w3id.org/ro/wfrun/process/0.1"}
```
[the remark this answers is not in the document]
Comment (stain) [reply docs/1.2-DRAFT/profiles.md]: I've modified this example to be complete of workflow/0.1 that then references both process/0.1 and workflow RO-Crate

File: docs/1.2-DRAFT/profiles.md [hunk 2]
```diff
@@ -118,6 +146,15 @@ try retrieving `https://about.workflowhub.eu/Workflow-RO-Crate/1.0/ro-crate-meta
...
 <!-- TODO Make both examples above actually work! -->
 
+#### Inheriting contextual entities from a Profile Crate
+
+If an RO-Crate declares conformance to a given profile crate with `conformsTo` on its root data entity, contextual entities declared in the corresponding Profile Crate do _not_ need to be repeated in the conforming crate. 
+
+For instance, if a Profile Crate adds a `DefinedTerm` entity according to the [ad-hoc definitions](appendix/jsonld.html#adding-new-or-ad-hoc-vocabulary-terms), the term can be referenced in the conforming crate without making a contextual entity there. For archival purposes it may however still be preferrable to copy such entities across to each conforming crate.
+
+It is recommended that `@id` of such inheritable entities are absolute URIs on both sides to avoid resolving relative paths, or that the profile recommends a [JSON-LD Context](#json-ld-context) to ensure consistent identifiers.
+
```
[the remark this answers is not in the document]
Comment (stain) [reply docs/1.2-DRAFT/profiles.md:181]: Yes, (finally) added such a recommendation:

> In the conforming crate, any terms defined in the profile using `DefinedTerm`, `rdfs:Class` and `rdf:Property` must be either be used as full URIs matching the `@id`, or mapped to these URIs from the conforming crate's JSON-LD `@context`. Note that JSON-LD only expands keys from `@id` and `@type`.

**Comment by stain** [comment]
I have finally managed to tidy up the profile section.

Changes:
- Merged to latest master (but pending #296 with retrieval section moved out from here)
- Use only `rdfs:Class` and `rdf:Property`, documented as _Schema.org style schemas_
- Inheritance across profiles now only "Multiple Profiles" as suggestions of conformance only
- Renamed section "Shared contextual entities from a Profile Crate"
- Examples expanded following @simleo's suggestion

**Comment by stain** [comment]
Suggestion in call 2024-04-25 @CaroleGoble was to also look at Delegation pattern:

- https://en.wikipedia.org/wiki/Delegation_pattern
- https://refactoring.guru/replace-delegation-with-inheritance
- https://stackoverflow.com/questions/832536/when-to-use-delegation-instead-of-inheritance
- https://www.jetbrains.com/help/idea/replace-inheritance-with-delegation.html

In a sense we use delegation pattern in https://trefx.uk/5s-crate/0.4/#referencing-a-workflow-crate as instead of subclassing Workflow RO-Crate, it references the Workflow RO-Crate by URL -- one crate to another.   I've documented this pattern in #296.

This would make the resulting 5s-crate as Workflow Run Crate not conform to any strict inheritance rules, as it would not have the usual `ComputationalWorkflow` etc. and so should not itself declare `conformsTo` the Workflow RO-Crate  (which we would have to then make Workflow RO-Crate indirection also be a workflow language!).  

Note a side issue: at the moment that 5s-crate is in that case purposefully ignoring the MUST of [conformsTo requirements](https://www.researchobject.org/workflow-run-crate/profiles/0.4/workflow_run_crate#requirements) -- we can argue over that separately!

File: docs/1.2-DRAFT/profiles.md [hunk 3]
```diff
@@ -89,32 +89,67 @@ It is valid for a crate to conform to multiple profiles, in which case `conforms
...
+    "name": "Process Run crate profile description",
+    "encodingFormat": "text/html",
+    "about": "https://w3id.org/ro/wfrun/process/0.4",
+},
+{ 
+    "@id": "#hasSpecification",
+    "@type": "ResourceDescriptor",
+    "hasRole": { "@id": "http://www.w3.org/ns/dx/prof/role/specification" },
+    "hasArtifact": {"@id": "index.html"}
 }
```
[the remark this answers is not in the document]
Comment (stain) [reply docs/1.2-DRAFT/profiles.md:143]: Added

**Comment by stain** [comment]
@simleo Good to go..?

**Review by simleo — APPROVED** [review]

**Comment by simleo** [comment]
> @simleo Good to go..?

AFAICT, yes

**Pull request merged**